contracts/integrations/contracts/test/TestFramework.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.5.9;
+
+
+contract TestFramework {
+
+    event SomeEvent(uint256 someNumber);
+
+    uint256 public counter;
+
+    function setCounter(uint256 newCounter)
+        external
+    {
+        counter = newCounter;
+    }
+
+    function revertSideEffect(uint256 returnValue)
+        external
+        returns (uint256)
+    {
+        if (counter != 0) {
```

Comment:
This kind of confuses me as to why we have these `counter` checks. Naively, I would expect this function to just revert always. What's the reason for this pattern?

Reply:
So, I think that it's worth mentioning that this contract does not have a purpose outside of testing the framework (primarily the "GetterCache"). With this in mind, this pattern intends to change the behavior of a given function---in this case it happens to be a revert function---through side effects so that the return value will be different even if the arguments are the same. This is helpful for testing the cache because we expect the old result to be in the cache rather than the new result. 

I'm all ears if there is a simpler way to test this pattern.